### PR TITLE
Update gaia build to go1.16

### LINKED ISSUE #812: Consider migrating build to go1.16
Evaluate downstream and upstream build and docker scripts, as well as github actions.

### LINKED ISSUE #836: Update build to use go 1.16
Tag PR with `sim` to test simulations too

### PR DESCRIPTION
**Pull request by shahankhatch**
Closes #836

### CODE REVIEW

**Comment by shahankhatch** [comment]
Need to alter `sim` label criteria in order to run long-running sim against this PR

**Comment by shahankhatch** [comment]
`SimLabeled / test-sim-nondeterminism-labeled` is now running. Will be RFR as soon as all checks pass, `SimLabeled / test-sim-nondeterminism-labeled` in particular.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/cosmos/gaia/v4
 
-go 1.15
+go 1.16
```
Comment (alessio) [inline go.mod:3]: You don't need to bump this

**Review by alessio — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (alessio) [reply go.mod:3]: I meant that the `go` tool automatically bumps this dependency when compiling if necessary

Comment (shahankhatch) [reply go.mod:3]: Thanks for your insight into go's implicit behaviour! 🙏 

I'll stick with 1.16 so that it's consistent and explicit.

**Pull request merged**